config/testutils/fake_cloud.go

```diff
@@ -42,10 +44,12 @@ type configAndCerts struct {
 }
 
 // NewFakeCloudServer creates and starts a new grpc server for the Viam Cloud.
-func NewFakeCloudServer(ctx context.Context, logger logging.Logger) (*FakeCloudServer, error) {
+func NewFakeCloudServer(t *testing.T, ctx context.Context, logger logging.Logger) (*FakeCloudServer, func()) { //revive:disable-line:context-as-argument
```

Comment:
side note, I tried to configure this rule globally (allowing `*testing.T` to come before `context.Context`) in `etc/.golangci.yaml` but was unsuccessful. If others agree / know how to do this please share!

Reply:
so I figured out how to configure this rule globally, but it has the side-effect of turning all other revive rules. furthermore, enable all rules explicitly seems to cause a panic due to [lack of defaults](https://github.com/mgechev/revive/issues/749) :disappointed:

Reply:
this turned into a rabbit hole... I think the problem i was describing goes away if we use a later version of revive, but there's another issue apparently: https://github.com/golangci/golangci-lint/issues/4353

not worth the hassle right now, just sticking with a `//nolint:revive`